### PR TITLE
Fix invalid email issue #69

### PR DESCRIPTION
**Pull request by andidev**

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
Thanks @andidev . Could you add a test to verify the improved behaviour?

**Comment by andidev** [comment]
@OlgaMaciaszek 
I have pushed some more tests + some refactoring to make testing easier + I have removed the CharConverter and replaced it with a static method in TextUtils and make sure to call the new stripAccent methods from all locales

File: src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import io.codearte.jfairy.producer.BaseProducer
+import io.codearte.jfairy.producer.VATIdentificationNumberProvider
+import io.codearte.jfairy.producer.company.CompanyProvider
+import spock.lang.Specification
+
+import static io.codearte.jfairy.producer.company.CompanyProvider.*
+
+class CompanyProviderSpec extends Specification {
+
+    def baseProducer = Mock(BaseProducer)
```
Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy]: I know this is completely subjective, but we are not big fans of "def"; could you use actual types, i.e. `BaseProducer baseProducer = Mock(BaseProducer)` . I think I have missed it and approved it in previous PRs, but I'm going to switch it. For the new one, please change it.

File: src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    def dataMaster = Mock(DataMaster)
+    def vatIdentificationNumberProvider = Mock(VATIdentificationNumberProvider)
+
+    def "should generate should remove slashes in company email"() {
```
Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy]: The name of this test should be modified (too many shoulds - select one thing you are testing)

File: src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    def "should strip accents from company email"() {
+        given:
+            baseProducer.trueOrFalse() >> false
+            dataMaster.getRandomValue(COMPANY_EMAIL) >> "info"
+            dataMaster.getRandomValue(COMPANY_NAME) >> "åäöéáąćęłńśóźż"
+            dataMaster.getRandomValue(DOMAIN) >> "com"
+
+        def companyProvider = new CompanyProvider(baseProducer, dataMaster, vatIdentificationNumberProvider);
+
```
Comment (OlgaMaciaszek) [inline src/test/groovy/io/codearte/jfairy/producer/company/locale/CompanyProviderSpec.groovy]: Please remove unnecessary empty lines in tests.

File: src/main/java/io/codearte/jfairy/producer/person/EmailProvider.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.google.inject.Provider;
+import io.codearte.jfairy.data.DataMaster;
+import io.codearte.jfairy.producer.BaseProducer;
+import io.codearte.jfairy.producer.util.TextUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import static io.codearte.jfairy.producer.person.PersonProvider.PERSONAL_EMAIL;
+import static org.apache.commons.lang3.StringUtils.lowerCase;
+
```
Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/person/EmailProvider.java]: +1

File: src/main/java/io/codearte/jfairy/producer/util/TextUtils.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2013. Codearte
+ */
+package io.codearte.jfairy.producer.util;
+
+import java.util.List;
+
+import static com.google.common.base.Joiner.on;
+
```
Comment (OlgaMaciaszek) [inline src/main/java/io/codearte/jfairy/producer/util/TextUtils.java]: +1

**Comment by andidev** [comment]
@OlgaMaciaszek 
So your remarks should be solved. I also converted from spaces to tabs in some files where I saw I hade used spaces instead of tabs.

**Pull request merged**